pkg/image/builder/image_builder.go

```diff
@@ -80,85 +81,196 @@ func (b *ImageBuilder) Run(ctx context.Context, targetRef string, layoutPath lay
 	if err != nil {
 		return err
 	}
-	idxManifest, err := idx.IndexManifest()
+	// make a copy of the original manifest for later
+	originalIdxManifest, err := idx.IndexManifest()
 	if err != nil {
 		return err
 	}
+	originalIdxManifest = originalIdxManifest.DeepCopy()
 
-	for _, manifest := range idxManifest.Manifests {
-		switch manifest.MediaType {
-		case types.DockerManifestSchema2:
-			v2format = true
-		case types.OCIManifestSchema1:
-			v2format = false
-		default:
-			return fmt.Errorf("image %q: unsupported manifest format %q", targetRef, manifest.MediaType)
-		}
-
-		img, err := layoutPath.Image(manifest.Digest)
+	/*
+		processImageIndex is a recursive function that allows for traversal of the hierarchy of
+		parent/child indexes that can exist for a multi arch image. There's always
+		at least one index at the root since this is an OCI layout that we're dealing with.
+		In theory there can be "infinite levels" of "index indirection" for multi arch images, but typically
+		its only two levels deep (i.e. index.json itself which is level one, and the manifest list
+		defined in the blobs directory, which is level two).
+	*/
+	var processImageIndex func(idx v1.ImageIndex) (v1.ImageIndex, error)
+	processImageIndex = func(idx v1.ImageIndex) (v1.ImageIndex, error) {
```

Comment:
same comment as before, is it possible to have it in a standalone function? Only because of readability purpose

Reply:
This one gets a little more complicated because the `v2format` value is updated along the way as needed. I'll have to include that as a pointer to the bool to maintain the same behavior.

Reply:
changed in commit https://github.com/openshift/oc-mirror/pull/611/commits/01c51f41b2b92ad160d3581da6f87cd930921050